src/WebJobs.Script.WebHost/Management/LinuxSpecialization/PackageCopyHandler.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using Microsoft.Azure.WebJobs.Script.Diagnostics;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Management.LinuxSpecialization
+{
+    public class PackageCopyHandler : IPackageCopyHandler
+    {
+        private readonly ILogger<PackageCopyHandler> _logger;
+
+        public PackageCopyHandler(ILogger<PackageCopyHandler> logger)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public string CopyPackageFile()
+        {
+            var home = Environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteHomePath);
+            var packageFolderPath = Path.Combine(home, "data", "SitePackages");
+
+            if (!Directory.Exists(packageFolderPath))
+            {
+                _logger.LogWarning($"{nameof(CopyPackageFile)} failed. SitePackages folder in the data folder doesn't exist.");
+                return string.Empty;
+            }
+
+            var packageNameTxtPath = Path.Combine(packageFolderPath, "packagename.txt");
+            if (!File.Exists(packageNameTxtPath))
+            {
+                _logger.LogWarning($"{nameof(CopyPackageFile)} failed. packagename.txt doesn't exist.");
+                return string.Empty;
+            }
+
+            var packageFileName = File.ReadAllText(packageNameTxtPath);
```

Comment:
>File [](http://example.com/codeflow?start=34&length=4)

for unittesting this- checkout FileUtility.Instance

Reply:
Added unit tests with `IFileSystem`.